internal/executor/service/job_lease.go

```diff
@@ -127,3 +138,25 @@ func (jobLeaseService *JobLeaseService) RenewJobLeases(jobs []*job.RunningJob) (
 
 	return failedJobs, nil
 }
+
+func getAvoidNodeLabels(pod *v1.Pod, avoidNodeLabelsOnRetry []string, clusterContext context2.ClusterContext) (map[string]string, error) {
+	if len(avoidNodeLabelsOnRetry) == 0 {
+		return map[string]string{}, nil
+	}
+
+	nodeName := pod.Spec.NodeName
+	if nodeName == "" {
+		return nil, errors.New("Empty node name in pod")
```

Comment:
Not sure this should be an error.

This would happen on webhook failures (i.e not never got submitted) OR even pod never got assigned to a node due to fragmentation so we are returning it.

Reply:
I've pushed another commit to improve log messages